### DISCUSSION TITLE
[Feature Request] Auto generated name for File storage

### DISCUSSION THREAD

**Discussion by zakaria-chahboun** [supabase/supabase#3150] [Feature Requests]
Hi everyone!
How i can upload a file in the storage without specifying a name for this file?

Thanks :heart_decoration:

**Comment by dshukertjr** [comment]
Hi @zakaria-chahboun!

You would have to specify the name of the file when you upload, so if you just want a file name that does not overlap with each other, you could use the current timestamp as the file name. You can also use [UUID](https://www.npmjs.com/package/uuid) package to generate uuid to be used as your file name if you want to be more sophisticated than just using timestamp as the file name!

**Comment by zakaria-chahboun** [comment]
@dshukertjr I know this methods. But supabase should have a native implementation for that (as a feature)

**Reply by dshukertjr** [reply]
@zakaria-chahboun In that case it might make sense to change the category of this question from "Q & A" to "Discussion".

**Reply by zakaria-chahboun** [reply]
done!
